functions/src/db.ts

```diff
@@ -1,15 +1,17 @@
 import * as admin from 'firebase-admin';
+import { readFileSync } from 'fs';
 import Database from 'common/firebase/database';
-import serviceAccount from './firebase-adminsdk.json';
 
 if (process.env.TEST_MODE) {
   // eslint-disable-next-line no-console
   console.log('We are in test mode. Skip initializing firebase-admin.');
-} else {
+} else if (process.env.DEV) {
   admin.initializeApp({
-    credential: admin.credential.cert(serviceAccount as admin.ServiceAccount),
+    credential: JSON.parse(readFileSync('./firebase-adminsdk.json').toString()),
     databaseURL: 'https://samwise-dev.firebaseio.com',
```

Comment:
The existing code is actually very problematic. For example, this line is probably the reason why some functions are not properly working in prod instance.